File: 2024/7/31/Q2304.py
N = int(input())
graph = []
maxL, maxH, maxIdx = 0, 0, 0
L, R, res = 0, 0, 0
for _ in range(N):
    a, b = map(int, input().split())
    graph.append([a, b])
    maxL = a if a > maxL else maxL
    maxH = b if b > maxH else maxH
    maxIdx = a if b == maxH else maxIdx
stk = [0 for _ in range(maxL + 1)]
for i, h in graph:
    stk[i] = h
for i in range(maxIdx + 1):
    L = stk[i] if stk[i] > L else L
    res += L
for i in range(maxL, maxIdx, -1):
    R = stk[i] if stk[i] > R else R
    res += R
print(res)

File: 2024/7/31/cpp/Q2304.cpp
#include<iostream>
#include<algorithm>
#include<vector>

using namespace std;

vector<pair<int, int>> graph;
vector<int> stk;
int maxL, maxH, maxIdx;

int main(){
    int n;
    int L = 0, R = 0, res = 0;
    cin >> n;
    for (int i = 0; i < n; i++){
        int l, h;
        cin >> l >> h;
        graph.push_back({l, h});
        maxL = maxL < l ? l : maxL;
        maxH = maxH < h ? h : maxH;
        maxIdx = maxH == h ? l : maxIdx;
    }
    stk.resize(maxL + 1);
    for (auto & g : graph){
        stk[g.first] = g.second;
    }
    for (int i = 0; i <= maxIdx; i++){
        L = L < stk[i] ? stk[i] : L;
        res += L;
    }
    
    for (int i = maxL; i > maxIdx; i--){
        R = R < stk[i] ? stk[i] : R;
        res += R;
    }
    cout << res << endl;

    return 0;
}